surrealist: 1.11.7 -> 2.0.6

MDM23 opened the pull request:
## Description of changes

Migrate to official new GitHub repository and update to the latest version.

- Closes: https://github.com/NixOS/nixpkgs/issues/306361#issue-2259624224
- [Changelog](https://github.com/surrealdb/surrealist/releases/tag/surrealist-v2.0.6)
- [Diff](https://github.com/surrealdb/surrealist/compare/v1.11.7...surrealist-v2.0.6)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

MDM23 committed:
surrealist: 1.11.7 -> 2.0.6

getchoo commented on pkgs/by-name/su/surrealist/package.nix:
```diff
@@ -35,123 +34,91 @@ let
-      # use --ignore-script and --no-optional to avoid downloading binaries
-      # use --frozen-lockfile to avoid checking git deps
-      pnpm install --frozen-lockfile --no-optional --ignore-script
+      pnpm install --frozen-lockfile --ignore-script
```
```suggestion
      # Some packages produce platform dependent outputs. We do not want to cache those in the global store
      pnpm config set side-effects-cache false
      # --force allows us to fetch all dependencies including ones that aren't meant for our host platform
      pnpm install --frozen-lockfile --ignore-script --force
```
this should remove the need for architecture specific outputHashes. see #290715

SuperSandro2000 commented on pkgs/by-name/su/surrealist/package.nix:
```diff
@@ -35,123 +34,89 @@ let
-      # use --ignore-script and --no-optional to avoid downloading binaries
-      # use --frozen-lockfile to avoid checking git deps
-      pnpm install --frozen-lockfile --no-optional --ignore-script
+      # Some packages produce platform dependent outputs. We do not want to cache those in the global store
+      pnpm config set side-effects-cache false
+      # --force allows us to fetch all dependencies including ones that aren't meant for our host platform
+      pnpm install --frozen-lockfile --ignore-script --force
```
Please update to use #290715

getchoo replied:
feel free to cherry pick from #318105 or rebase when/if it's merged

MDM23 replied:
@getchoo Thank you, will check it out once merged!

MDM23 commented:
Rebased and updated to v2.0.6 for now. Version 2.1 is built using tauri 2 beta and I could not yet find out how to override [this dependency](https://github.com/NixOS/nixpkgs/blob/57d6973abba7ea108bac64ae7629e7431e0199b6/pkgs/development/tools/rust/cargo-tauri/default.nix#L20).
